backend/clubs/views.py

```diff
@@ -481,8 +481,19 @@ def filter_queryset(self, request, queryset, view):
         return new_queryset
 
 
+def get_officers(event):
+    officers = []
+    for visit in ZoomMeetingVisit.objects.filter(event=event, leave_time=None):
+        m = Membership.objects.get(club=event.club, person=visit.person)
+        if m and m.role <= Membership.ROLE_OFFICER:
+            officers.append(visit.person.username)
```

Comment:
This is extremely inefficient and will result in `(# of events) * (# of visits) * 2` SQL queries in the worst case. Given ~600 events and ~10 visits per event, this is 1,200 SQL queries per request.

You can get this down to 1 SQL query with a carefully crafted query.

Reply:
Ah, typoed here, it would actually be 12,000 SQL queries.

Reply:
Ok let me probe